types/react/index.d.ts

```diff
@@ -184,10 +184,10 @@ declare namespace React {
     // Component API
     // ----------------------------------------------------------------------
 
-    type ReactInstance = Component<any, any> | Element;
+    type ReactInstance = Component<any> | Element;
 
     // Base component for plain JS classes
-    interface Component<P, S> extends ComponentLifecycle<P, S> { }
+    interface Component<P, S = {}> extends ComponentLifecycle<P, S> { }
```

Comment:
Does `null` work? I don't think React initializes an empty state object for you.